src/test/java/seedu/address/logic/commands/AddShortcutCommandTest.java

```diff
@@ -0,0 +1,296 @@
+        AddShortcutCommand addShortcutCommand = new AddShortcutCommand(validShortcutName, validShortcutCommand);
+        AddShortcutCommandTest.ModelStub modelStub = new AddShortcutCommandTest.ModelStubWithShortcut(validShortcut);
+
+        assertThrows(CommandException.class, AddShortcutCommand.MESSAGE_DUPLICATE_SHORTCUT, () ->
+                addShortcutCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        Shortcut first = new Shortcut("first", "list");
+        Shortcut second = new Shortcut("second", "help");
+        AddShortcutCommand addFirstShortcutCommand =
+                new AddShortcutCommand(first.getShortcutName(), first.getShortcutCommand());
+        AddShortcutCommand addSecondShortcutCommand =
+                new AddShortcutCommand(second.getShortcutName(), second.getShortcutCommand());
+
+        // same object -> returns true
+        assertTrue(addFirstShortcutCommand.equals(addFirstShortcutCommand));
+
+        // same values -> returns true
+        AddShortcutCommand addFirstShortcutCommandCopy =
+                new AddShortcutCommand(first.getShortcutName(), first.getShortcutCommand());
+        assertTrue(addFirstShortcutCommand.equals(addFirstShortcutCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addFirstShortcutCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addFirstShortcutCommand.equals(null));
+
+        // different shortcut -> returns false
+        assertFalse(addFirstShortcutCommand.equals(addSecondShortcutCommand));
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
+
+        @Override
+        public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyUserPrefs getUserPrefs() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public GuiSettings getGuiSettings() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setGuiSettings(GuiSettings guiSettings) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public Path getAddressBookFilePath() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setAddressBookFilePath(Path addressBookFilePath) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public Path getShortcutLibraryFilePath() {
+            return null;
+        }
+
+        @Override
+        public void setShortcutLibraryFilePath(Path shortcutLibraryFilePath) {
+
+        }
```

Comment:
Another empty method where you might want to throw AssertionError if they are not supposed to be called.